src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -390,38 +378,27 @@ private DateTime(SerializationInfo info, StreamingContext context)
                 switch (enumerator.Name)
                 {
                     case TicksField:
-                        serializedTicks = Convert.ToInt64(enumerator.Value, CultureInfo.InvariantCulture);
+                        _dateData = (ulong)Convert.ToInt64(enumerator.Value, CultureInfo.InvariantCulture);
                         foundTicks = true;
-                        break;
+                        continue;
                     case DateDataField:
-                        serializedDateData = Convert.ToUInt64(enumerator.Value, CultureInfo.InvariantCulture);
-                        foundDateData = true;
-                        break;
-                    default:
-                        // Ignore other fields for forward compatibility.
-                        break;
+                        _dateData = Convert.ToUInt64(enumerator.Value, CultureInfo.InvariantCulture);
+                        goto foundData;
                 }
             }
-            if (foundDateData)
-            {
-                _dateData = serializedDateData;
-            }
-            else if (foundTicks)
-            {
-                _dateData = (ulong)serializedTicks;
-            }
-            else
+            if (!foundTicks)
             {
                 throw new SerializationException(SR.Serialization_MissingDateTimeData);
             }
-            long ticks = InternalTicks;
-            if (ticks < MinTicks || ticks > MaxTicks)
+        foundData:
+            if (UTicks > MaxTicks)
             {
                 throw new SerializationException(SR.Serialization_DateTimeTicksOutOfRange);
             }
         }
 
         internal long InternalTicks => (long)(_dateData & TicksMask);
+        private ulong UTicks => _dateData & TicksMask;
```

Comment:
Why create `UTicks` instead of just making `InternalTicks ` a `ulong`?

Also I'm not sure why `InternalTicks` is `internal` anyway - it doesn't appear used outside this file.  Shouldn't it be `private`?

Reply:
Don't know why there was `InternalTicks` at all - it was exactly the same as `Ticks`. Removed it and replaced all uses with `Ticks`.

Reply:
That is make sense.